app/models/orphan.rb

```diff
@@ -96,16 +109,12 @@ def eligible_for_sponsorship?
     Orphan.active.currently_unsponsored.include? self
   end
 
-  def currently_sponsored?
-    sponsorships.all_active.present?
-  end
-
   def current_sponsorship
-    sponsorships.all_active.first if currently_sponsored?
+    sponsorships.all_active.first if sponsored?
```

Comment:
It's also a possibility but maybe not recommended to have a task that runs weekly that checks all the `sponsorship_status` against that.